test/endtoendtestutils.cpp

```diff
@@ -0,0 +1,167 @@
+        QByteArray credHash = QByteArray(_cred->user().toUtf8() + ":" + _cred->password().toUtf8()).toBase64();
+        req.setRawHeader("Authorization", "Basic " + credHash);
+
+        return OCC::AccessManager::createRequest(op, req, outgoingData);
+    }
+
+private:
+    // The credentials object dies along with the account, while the QNAM might
+    // outlive both.
+    QPointer<const EndToEndTestCredentials> _cred;
+};
+
+/** End to end test credentials class **/
+
+QNetworkAccessManager *EndToEndTestCredentials::createQNAM() const
+{
+    return new EndToEndTestCredentialsAccessManager(this);
+}
+
+/** End to end test helper class **/
+
+EndToEndTestHelper::~EndToEndTestHelper()
+{
+    removeConfiguredSyncFolder();
+    removeConfiguredAccount();
+
+    OCC::AccountManager::instance()->shutdown();
+}
+
+void EndToEndTestHelper::startAccountConfig()
+{
+    const auto accountManager = OCC::AccountManager::instance();
+    _account = accountManager->createAccount();
+
+    _account->setCredentials(new EndToEndTestCredentials);
+    _account->setUrl(OCC::Theme::instance()->overrideServerUrl());
+
+    const auto serverUrlString = QString(serverUrl);
+    _account->setUrl(serverUrlString);
+
+    _account->networkAccessManager()->setProxy(QNetworkProxy(QNetworkProxy::NoProxy));
+    _account->setSslConfiguration(QSslConfiguration::defaultConfiguration());
+    _account->setSslErrorHandler(new OCC::SimpleSslErrorHandler);
+    _account->setTrustCertificates(true);
+
+    slotConnectToNCUrl(serverUrlString);
+}
+
+void EndToEndTestHelper::slotConnectToNCUrl(const QString &url)
+{
+    qDebug() << "Connect to url: " << url;
+
+    const auto fetchUserNameJob = new OCC::JsonApiJob(_account->sharedFromThis(), QStringLiteral("/ocs/v1.php/cloud/user"));
+    connect(fetchUserNameJob, &OCC::JsonApiJob::jsonReceived, this, [this, url](const QJsonDocument &json, const int statusCode) {
+        if (statusCode != 100) {
+            qDebug() << "Could not fetch username.";
+        }
+
+        const auto objData = json.object().value("ocs").toObject().value("data").toObject();
+        const auto userId = objData.value("id").toString("");
+        const auto displayName = objData.value("display-name").toString("");
+        _account->setDavUser(userId);
+        _account->setDavDisplayName(displayName);
+
+        _accountState = new OCC::AccountState(_account);
+
+        emit accountReady(_account);
+    });
+    fetchUserNameJob->start();
+}
+
+void EndToEndTestHelper::removeConfiguredAccount()
+{
+    OCC::AccountManager::instance()->deleteAccount(_accountState.data());
+}
+
+OCC::Folder *EndToEndTestHelper::configureSyncFolder()
+{
+    if(_syncFolder) {
+        removeConfiguredSyncFolder();
+    }
+
+    qDebug() << "Creating temp end-to-end test folder.";
+    Q_ASSERT(_tempDir.isValid());
+    OCC::FileSystem::setFolderMinimumPermissions(_tempDir.path());
+    qDebug() << "Created temp end-to-end test folder at:" << _tempDir.path();
```

Comment:
please create a logging category and make use of it
that will allow filtering and easy retrieving during analysis